compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -782,9 +782,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
 
         match (source.kind(), target.kind()) {
             // Trait+Kx+'a -> Trait+Ky+'b (upcasts).
-            (&ty::Dynamic(ref data_a, _, dyn_a), &ty::Dynamic(ref data_b, _, dyn_b))
-                if dyn_a == dyn_b =>
```

Comment:
Surprised by this removal

Reply:
I see the more specific pattern here and in the next file, but doesn't this code have to run for non `dyn*`?

Reply:
Can you be more specific? Did you see the PR description that mentions "Also, disable dyn* trait upcasting"?

Because all this disables is `CoerceUnsized` implementations from `dyn* Trait` to `dyn* SuperTrait`, which cannot be implemented with `CoerceUnsized` and need to be reworked.

Reply:
I see. r=me after rebasing